pkg/ebpf/c/tracee.bpf.c

```diff
@@ -1854,38 +1860,27 @@ static __always_inline task_info_t *init_task_info(u32 tid, u32 pid, bool *initi
         *initialized = task_info == NULL;
     }
     if (unlikely(task_info == NULL)) {
-        // get the submit buffer to fill the task_info_t in the map
-        // this is done because allocating the stack space for task_info_t usually
-        // crosses the verifier limit.
-        // todo: use scratch map both for proc_info and thread_info to avoid using submit_buffer
-        int buf_idx = SUBMIT_BUF_IDX;
-        void *submit_buffer = bpf_map_lookup_elem(&bufs, &buf_idx);
-        if (unlikely(submit_buffer == NULL))
-            return NULL;
+        int zero = 0;
 
-        bpf_map_update_elem(&task_info_map, &tid, submit_buffer, BPF_NOEXIST);
-        task_info = bpf_map_lookup_elem(&task_info_map, &tid);
-        // appease the verifier
-        if (unlikely(task_info == NULL)) {
+        scratch_t *scratch = bpf_map_lookup_elem(&scratch_map, &zero);
+        if (unlikely(scratch == NULL))
             return NULL;
-        }
+
         proc_info_t *proc_info = bpf_map_lookup_elem(&proc_info_map, &pid);
         if (proc_info == NULL) {
-            bpf_map_update_elem(&proc_info_map, &pid, submit_buffer, BPF_NOEXIST);
-            proc_info = bpf_map_lookup_elem(&proc_info_map, &pid);
-            // appease the verifier
-            if (unlikely(proc_info == NULL)) {
-                return NULL;
-            }
+            scratch->proc_info.new_proc = false;
```

Comment:
Since we use the same scratch for the `proc_info_t` and the `task_info_t` as a union, if we enter this branch isn't it possible the `task_info_t` data will be then corrupted by the data set before?

Reply:
No. 
We set all proc fields here, then update proc_info with the data.
Then we set all task fields and update the task_info map.
Since we set all fields of the struct, there shouldn't be any problem with that.